### PR TITLE
chore(tui): polish, stability, docs, and v0.2.0 release

### LINKED ISSUE #23: TUI: полировка, устойчивость, документация, релиз 0.2.0
> **Обязательный контекст:** перед работой прочитать `docs/DESIGN_PHILOSOPHY.md`
> в корне репозитория — все решения принимаются в его рамках.
> Это задача 11 из 11 плана `docs/TUI_MODERNIZATION_PLAN.md`.
> **Зависит от:** задача 9 «Выделение текста мышью и копирование в буфер», задача 10 «Мышь в интерактивном режиме терминала» — не начинать, пока их PR не смержены в main.
>
---

# Полировка, устойчивость, документация

**Шаги:**

1. **Стабильность layout:** пройтись по всем режимам — высоты зон фиксированы
   или меняются плавно (рост поля ввода — единственное легитимное изменение).
   Убрать `terminal.clear()` на каждое agent-событие (`needs_clear`), если после
   редизайна артефактов нет — полный clear вызывает мерцание; оставить только
   на смену режима.
2. **Ресайз:** сжать окно до 40×10 — ничего не паникует (все `saturating_sub`
   на месте, модал клампится, кэш перестраивается).
3. **Деградация:** запуск в терминале без поддержки мыши — приложение работает
   как раньше (ошибки включения mouse capture игнорировать с `warn!`).
4. **Help-бар** обновить под новые возможности (упомянуть мышь: `wheel scroll`,
   `click expand`, `drag copy` — коротко, в 1 строку, лишнее убрать).
5. **Тесты:** прогнать полный набор `cargo test --workspace`; добавить
   недостающие unit-тесты (hit_test, SSE-парсер, markdown-lite, кламп скролла).
6. **Документация:** обновить `README.md` (Features: Mouse support, Streaming
   responses; новый скриншот-плейсхолдер) и `USER_GUIDE.md` (раздел «Мышь»:
   скролл, клики, выделение/копирование; изменение — Enter в диалоге
   подтверждения активирует выделенную кнопку, дефолт Deny).
7. `CHANGELOG`/`PROGRESS.md` — записать итог, поднять версию до `0.2.0`
   в `workspace.package`.

**DoD:**
- `cargo build --release`, `cargo clippy --all-targets -- -D warnings`,
  `cargo test --workspace` — зелёные.
- Ручной smoke-тест сценария: запуск → вопрос агенту → стриминг ответа →
  подтверждение команды кликом → разворот вывода кликом → выделение и копия
  текста → Ctrl+T → скролл колесом → Ctrl+T назад → Ctrl+C → сессия сохранена.
- Документация отражает новые возможности.

---

---

## Критерии готовности (общие, в дополнение к DoD выше)

- [ ] `cargo build --workspace` зелёный
- [ ] `cargo clippy --all-targets -- -D warnings` зелёный
- [ ] `cargo test --workspace` зелёный
- [ ] `PROGRESS.md` обновлён (что сделано, принятые решения, изменения контрактов)
- [ ] Поведение вне скоупа задачи не изменилось (хоткеи, `!`, Ctrl+T/P/C, секреты, сессии)

### PR DESCRIPTION
**Pull request by devlawey**
Closes #23

## Summary

Final polish pass for the **0.2.0 milestone release** — stability fixes, documentation, tests, and version bump.

### Changes

- **Layout stability**: Removed `needs_clear` on non-TextDelta agent events that caused screen flickering. Full `terminal.clear()` now only fires on mode change.
- **Mouse capture degradation**: Split `EnableMouseCapture` from `EnterAlternateScreen`. If mouse capture fails (e.g. terminal without mouse support), the error is logged with `warn!` instead of crashing — the app continues with keyboard-only input.
- **Help-bar updates**: Normal mode now shows `wheel scroll`, `click expand`, `drag copy`. Thinking mode shows `wheel scroll` instead of `pgup/pgdn`.
- **9 new unit tests**:
  - Scroll clamp edge cases (4): zero when content fits, zero height no panic, exact fit, tiny terminal hit-test
  - Markdown-lite (3): empty string, only markers, multiple code spans
  - SSE parser (2): malformed data line, partial chunk
- **Documentation**: Updated README.md (Mouse Support + Streaming Responses in Features, updated keyboard shortcuts) and USER_GUIDE.md (new section 4.2 "Управление мышью" with full action table, renumbered sections).
- **Version**: Bumped from `0.1.1` → `0.2.0`.

### Tests

- `cargo clippy --workspace` — clean, no warnings
- `cargo test --workspace` — **242 passed, 0 failed, 5 ignored**
- Ignored tests require docker-sshd container (manual verification):
  - `transport::ssh::tests::*` (5 tests)

### Out of scope

- No public trait changes (`CommandExecutor`, `LlmClient` untouched)
- No remote-side changes (zero-install invariant preserved)
- Release publish (GitHub Release + binary build) will be done separately via `/prepare-release`

## Summary by CodeRabbit

* **New Features**
  * Added terminal mouse support: scroll wheel navigation, click-to-expand command output, and drag-to-select/copy text, with updated help and keyboard shortcuts.
  * Enhanced live/streaming responses with improved loading feedback.
* **Bug Fixes**
  * Improved UI stability during streaming updates and terminal resizing; reduced unnecessary full-screen clears.
  * Made mouse features gracefully degrade when mouse capture isn’t available.
  * Addressed edge cases across scrolling, hit-testing, markdown-lite rendering, and SSE streaming parsing.
* **Documentation**
  * Updated README and user guide with revised shortcuts, new mouse controls, and release version 0.2.0.

### CODE REVIEW

File: crates/agent/src/glm.rs [hunk 1]
```diff
@@ -1075,4 +1075,32 @@ data: {\"choices\":[{\"delta\":{\"content\":\" world\"}}]}
...
+        let mut state = SseState::new();
+        state.process_chunk("data: {\"choices\":[{\"delta\":{\"content\":\"Hi\"}}]}\n");
+        // Not yet complete — no response available.
+        state.process_chunk("\n");
+        let response = state.into_response().unwrap();
+        match response {
+            ChatResponse::Text(text) => assert_eq!(text, "Hi"),
+            _ => panic!("expected Text response"),
+        }
+    }
```
[the remark this answers is not in the document]
Comment (devlawey) [reply crates/agent/src/glm.rs:1105]: Fixed in f5715fd. The test now splits the SSE line across two chunks — the first chunk has no line terminator, and we assert `d1.is_empty()` before the second chunk completes the line.

File: crates/tui/src/runner.rs [hunk 2]
```diff
@@ -142,8 +142,13 @@ pub async fn run(
...
     let mut stdout = io::stdout();
-    crossterm::execute!(stdout, EnterAlternateScreen, EnableMouseCapture)
-        .map_err(|e| CoreError::Other(format!("failed to enter alternate screen: {e}")))?;
+    crossterm::execute!(stdout, EnterAlternateScreen).map_err(|e| {
+        CoreError::Other(format!("failed to enter alternate screen: {e}"))
+    })?;
+    // Mouse capture is optional — degrade gracefully if unsupported.
+    if let Err(e) = crossterm::execute!(io::stdout(), EnableMouseCapture) {
+        warn!(error = %e, "mouse capture not available — mouse support disabled");
+    }
```
[the remark this answers is not in the document]
Comment (devlawey) [reply crates/tui/src/runner.rs:153]: Fixed in f5715fd. Added `disable_raw_mode().ok()` on the error path before returning, so the terminal is restored if `EnterAlternateScreen` fails.

**Comment by devlawey** [comment]
All 3 review comments addressed in f5715fd:

1. **`sse_parse_partial_chunk`** — Rewritten to split the SSE line across two chunks without a trailing newline on the first piece. Added `assert!(d1.is_empty())` to verify the intermediate state.

2. **Raw mode cleanup on `EnterAlternateScreen` failure** — Added `disable_raw_mode().ok()` on the error path before returning, so the terminal is restored.

3. **`hit_test_tiny_terminal` nitpick** — Populated `layout_cache.lines` with 3 `RenderedLine` entries and tightened the assertion to `HitZone::Chat { .. }` (was `ChatEmpty | Chat`), so the test now meaningfully exercises the `Chat` branch.

All tests green: 242 passed, 0 failed, 5 ignored.
